JDK12 & JDK13 test failures

Mostly about classloading and powermock.

http://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-3.x-ZuluJDK13/3/testReport/

**Stacktrace**
```
java.lang.RuntimeException: java.lang.NoSuchFieldException: parent
	at com.hazelcast.util.FilteringClassLoader.<init>(FilteringClassLoader.java:59)
	at com.hazelcast.client.usercodedeployment.ClientUserCodeDeploymentTest.createNodeConfig(ClientUserCodeDeploymentTest.java:87)
	at com.hazelcast.client.usercodedeployment.ClientUserCodeDeploymentTest.testWithMultipleNodes_clientReconnectsToNewNode(ClientUserCodeDeploymentTest.java:142)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:114)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:835)
Caused by: java.lang.NoSuchFieldException: parent
	at java.base/java.lang.Class.getDeclaredField(Class.java:2417)
	at com.hazelcast.util.FilteringClassLoader.<init>(FilteringClassLoader.java:53)
	... 14 more
```

--- a/hazelcast/src/test/java/com/hazelcast/util/FilteringClassLoader.java
+++ b/hazelcast/src/test/java/com/hazelcast/util/FilteringClassLoader.java
@@ -21,7 +21,6 @@ import com.hazelcast.spi.annotation.PrivateApi;
 import java.io.ByteArrayOutputStream;
 import java.io.IOException;
 import java.io.InputStream;
-import java.lang.reflect.Field;
 import java.net.URL;
 import java.util.Collections;
 import java.util.Enumeration;
@@ -46,15 +45,12 @@ public class FilteringClassLoader extends ClassLoader {
     private ClassLoader delegatingClassLoader;
 
     public FilteringClassLoader(List<String> excludePackages, String enforcedSelfLoadingPackage) {
+        super(null);
         this.excludePackages = excludePackages;
         this.enforcedSelfLoadingPackage = enforcedSelfLoadingPackage;
 
         try {
-            Field parent = ClassLoader.class.getDeclaredField("parent");
-            parent.setAccessible(true);
-
-            delegatingClassLoader = (ClassLoader) parent.get(this);
-            parent.set(this, null);
+            delegatingClassLoader = ClassLoader.getSystemClassLoader();
         } catch (Exception e) {
             throw new RuntimeException(e);
         }

